crates/rari-doc/src/translations.rs

```diff
@@ -82,10 +84,26 @@ pub(crate) fn get_other_translations_for(slug: &str, locale: Locale) -> Vec<(Loc
         Locale::for_generic_and_spas()
             .iter()
             .filter_map(|l| {
-                Doc::page_from_slug(slug, *l, false)
-                    .ok()
-                    .map(|d| (*l, d.title().to_string()))
+                if slug.contains("/docs/") {
+                    Doc::page_from_slug(slug, *l, false).ok()
+                } else {
+                    SPA::from_slug(slug, *l)
+                }
```

Comment:
Similarly, `Doc::page_from_slug()` only returns doc pages.

@fiji-flo Is there something like `Page::from_slug()` that would get the doc no matter what it is?

Reply:
@fiji-flo Is there any reason I shouldn't use `Page::from_url()` directly?

See: 9cc1e04

Reply:
Yes, we're trying to avoid spreading the logic of the url layout to much and we allocate the String for the url just to take it apart again.

If you don't want to move it to the trait you can use:

```
   Page::internal_from_url(
        url: &str,
        locale: Option<Locale>,
        fallback: bool,
    ) -> Result<Self, DocError>
```
You just want to change the handling for Blog and Curriculim in the implementation (return NotFound for any non en-US locale).

But this still causes a lot of unnecessary look-ups. Moving this to the traits is the best way.

Reply:
Can you outline what you mean by moving this to the traits?

Reply:
in https://github.com/mdn/rari/blob/main/crates/rari-doc/src/pages/page.rs#L255

add 
```
    fn get_other_translations(&self) -> Vec<Translation>;
```

And then implement it for all implementations of PageLike.